crates/fuel-core/src/graphql_api/storage.rs

```diff
@@ -58,20 +58,22 @@ pub mod transactions;
 pub enum Column {
     /// The column id of metadata about the blockchain
     Metadata = 0,
+    /// Metadata for genesis progress
+    GenesisMetadata = 1,
```

Comment:
Do we need a separate column? Maybe we can use `Metadata` column instead?

Reply:
Would cause concurrency issues. The metadata is sa `() -> Metadata` mapping. All threads would then need to update the same key. Since we must always write the whole `OwnedValue` that could cause one genesis worker to overwrite another if we don't sync the access between them.

If you have another solution or feel that a mutex is acceptable if it means getting rid of the `GenesisMetadata` table then please write and I'll tackle it in another PR.